packages/safe-tools-client/app/utils/chain-connection-manager.ts

```diff
@@ -461,23 +493,27 @@ class WalletConnectConnectionStrategy extends ConnectionStrategy {
       };
     }
 
-    const hubConfigApi = new HubConfig(config.hubUrl);
-    const hubConfigResponse = await hubConfigApi.getConfig();
-
-    const provider = new WalletConnectProvider({
-      chainId,
-      infuraId: config.infuraId,
-      rpc: {
-        [networkIds[this.networkSymbol]]:
-          hubConfigResponse.web3.ethereum.rpcNodeHttpsUrl, // FIXME don’t hardcode ethereum
-      },
-      rpcWss: {
-        [networkIds[this.networkSymbol]]:
-          hubConfigResponse.web3.ethereum.rpcNodeWssUrl,
-      },
-      // based on https://github.com/WalletConnect/walletconnect-monorepo/blob/7aa9a7213e15489fa939e2e020c7102c63efd9c4/packages/providers/web3-provider/src/index.ts#L47-L52
-      connector: new CustomStorageWalletConnect(connectorOptions, chainId),
-    });
+    let provider;
+    if (this.mockProvider) {
+      provider = this.mockProvider as TestWalletConnectProvider;
+    } else {
```

Comment:
Maybe we could take leverage of the Environments, and if it's the test env we skip the walletConnect one